File: #0645.set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {

        int duplication = -1;
        for (int n : nums) {
            if (nums[abs(n) - 1] < 0) {
                duplication = abs(n);
            } else {
                nums[abs(n) - 1] *= -1;
            }
        }

        int size = nums.size();
        int missing = -1;
        for (int i = 0; i < size; i++) {
            if (nums[i] > 0) {
                missing = i + 1;
            }
        }
        return vector<int> {duplication, missing};
    }
};

File: #0645.set-mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    let arr = [-1, -1];
    let map = {};
    for (let num of nums) {
        if (num in map) {
           arr[0] = num; 
        }
        map[num] = num;
    }
    
    let length = nums.length;
    for (let i = 1; i <= length; i++) {
        if (!(i in map)) {
            arr[1] = i;
            break;
        }
    }
    return arr;
};
